src/runtime_action.rs

```diff
@@ -131,37 +126,57 @@ impl RuntimeAction {
         }
     }
 
-    pub fn process_request(&self) -> RequestResult {
+    pub fn build_request(&self) -> RequestResult {
         match self.conditions_apply() {
             Ok(false) => Ok(None),
             Ok(true) => match self.build_message() {
                 Ok(message) => Ok(self.grpc_service().build_request(message)),
-                Err(_) => self.resolve_failure_mode().map(|_| None),
+                Err(e) => match self.get_failure_mode() {
+                    FailureMode::Deny => Err(e),
+                    FailureMode::Allow => {
+                        debug!("continuing as FailureMode Allow");
+                        Ok(None)
+                    }
+                },
+            },
+            Err(e) => match self.get_failure_mode() {
+                FailureMode::Deny => Err(e.into()),
+                FailureMode::Allow => {
+                    debug!("continuing as FailureMode Allow");
+                    Ok(None)
+                }
             },
-            Err(_) => self.resolve_failure_mode().map(|_| None),
         }
     }
 
     pub fn process_response(&self, msg: &[u8]) -> ResponseResult {
-        match self {
-            Self::Auth(auth_action) => match Message::parse_from_bytes(msg) {
-                Ok(check_response) => auth_action.process_response(check_response),
-                Err(e) => {
-                    debug!("process_response(auth): failed to parse response `{e:?}`");
-                    self.resolve_failure_mode()
-                }
-            },
-            Self::RateLimit(rl_action) => match Message::parse_from_bytes(msg) {
-                Ok(rate_limit_response) => rl_action.process_response(rate_limit_response),
-                Err(e) => {
-                    debug!("process_response(rl): failed to parse response `{e:?}`");
-                    self.resolve_failure_mode()
+        let res = match self {
+            Self::Auth(auth_action) => {
+                let check_response =
+                    Message::parse_from_bytes(msg).map_err(ProcessGrpcMessageError::from)?;
+                auth_action.process_response(check_response)
+            }
+            Self::RateLimit(rl_action) => {
+                let rate_limit_response =
+                    Message::parse_from_bytes(msg).map_err(ProcessGrpcMessageError::from)?;
```

Comment:
Same: No need to `map_err` here, `?` will `From` (see [here](https://github.com/rust-lang/rust/issues/31436#issuecomment-299638299)) automagically.